File: Part 2 -  Regression/Section 4 - Simple Linear Regression/regresionLinealSimple.py
# -*- coding: utf-8 -*-
"""
Created on Wed May 18 12:12:22 2022

@author: Abigail
"""

#Regresión lineal simple

#Cómo importar las librerías

import numpy as np #matemáticas
import matplotlib.pyplot as plt #graficas y representación visual
import pandas as pd #carga de datos

#importar el dataset
dataset = pd.read_csv('Salary_Data.csv')
#iloc sirve para localizar los elemntos (filas y columnas) de un sataset
#por posicion i=index
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values


#Dividir el data set en conjunto d eentrenamiento y conjunto de test
#size=0.2 implica que un 20% de los datos se destinaran a testing, max0.3
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=1/3, random_state=0)



#Escalado de variables 
#Se hace a menudo pero no siempre
#Para el caso de la regresión lineal simple no se necesita escalar 
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
#solo transform para que detecte la transformaci+on que se hizo en el de entrenamiento
X_test = sc_X.transform(X_test)"""


#Crear el modelo de regresión lineal simple con el conjunto de ENTRENAMIENTO
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(X_train, y_train)

#Predecir el conjunto de TEST 
y_pred = regression.predict(X_test)

## Visualizar los resultados de ENTRENAMIENTO
plt.scatter(X_train, y_train, color="red")
plt.plot(X_train, regression.predict(X_train), color="blue")
plt.title("Suelvo vs Años de experiencia (Conjunto de entrenamiento)")
plt.xlabel("Años de experiencia")
plt.ylabel("Sueldo (US)")

## Visualizar los resultados de TEST
#la recta es la misma para ambos grupos de datos, lo que cambia son los puntos
plt.scatter(X_test, y_test, color="green")
plt.plot(X_train, regression.predict(X_train), color="blue")
plt.title("Suelvo vs Años de experiencia (Conjunto de test)")
plt.xlabel("Años de experiencia")
plt.ylabel("Sueldo (US)")

File: Part 2 -  Regression/Section 4 - Simple Linear Regression/regresionLinealSimple.R

#Regresión lineal simple
#Importar el data set

dataset = read.csv('Salary_Data.csv')


#Dividir los datos en conjunto de entrenamiento 
install.packages("caTools")

library(caTools)
set.seed(123) #semilla aleatoria
#0.8 es el porcentaje que sive para entrenar (train)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)


#Ajustar el modelo de regresión lineal simple con el conjunto de ENTRENAMIENTO  
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

summary(regressor)

#Ajustar el modelo de regresión lineal con el conjunto de TEST
y_pred = predict(regressor, newdata = testing_set)


#Visualización de los resultados en el conjunto de ENTRENAMIENTO
install.packages("ggplot2")
library(ggplot2)

ggplot()+
  geom_point(aes(x= training_set$YearsExperience, 
                 y = training_set$Salary),
             colour="red") +
  geom_line(aes(x =training_set$YearsExperience, 
                y= predict(regressor, newdata = training_set)),
              colour="blue") +
  ggtitle("Sueldo vs Años de experiencia (Conjunto de entrenamiento)") +
  xlab("Años de experiencia") +
  ylab("Sueldo en US")



#Visualización de los resultados en el conjunto de TEST
install.packages("ggplot2")
library(ggplot2)

ggplot()+
  geom_point(aes(x= testing_set$YearsExperience, 
                 y = testing_set$Salary),
             colour="black") +
  geom_line(aes(x =training_set$YearsExperience, 
                y= predict(regressor, newdata = training_set)),
            colour="blue") +
  ggtitle("Sueldo vs Años de experiencia (Conjunto de testing)") +
  xlab("Años de experiencia") +
  ylab("Sueldo en US")








